homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,187 @@
+        _LOGGER.debug(
+            "Still image at '%s' detected format: %s", info[CONF_STILL_IMAGE_URL], fmt
+        )
+        if fmt is None:
+            return False, "invalid_still_image"
+
+        # Second level functionality is to get a stream.
+        if not (stream_source := info.get(CONF_STREAM_SOURCE)):
+            return True, ""
+        try:
+            stream_source = info[CONF_STREAM_SOURCE]
+            # For RTSP streams, prefer TCP. This code is duplicated from
+            # homeassistant.components.stream.__init__.py:create_stream()
+            # It may be possible & better to call create_stream() directly.
+            stream_options: dict[str, str] = {}
+            if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
+                stream_options = {
+                    "rtsp_flags": "prefer_tcp",
+                    "stimeout": "5000000",
+                }
+            if rtsp_transport := info.get(CONF_RTSP_TRANSPORT):
+                stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = rtsp_transport
+            container = await self.hass.async_add_executor_job(
+                partial(
+                    av.open,
+                    stream_source,
+                    options=stream_options,
+                    timeout=SOURCE_TIMEOUT,
+                )
+            )
+            _ = container.streams.video[0]
+            return True, ""
+        except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+            return False, "stream_unauthorised"
+        except (KeyError, IndexError):
+            return False, "stream_novideo"
+        except PermissionError:
+            return False, "stream_not_permitted"
+        except OSError as err:
+            if "No route to host" in str(err):
+                return False, "stream_no_route_to_host"
+            if "Input/output error" in str(err):
+                return False, "stream_io_error"
+            raise err
+
+    @staticmethod
+    def build_schema(user_input):
+        """Create schema for camera config setup."""
+        spec = {
+            vol.Optional(CONF_NAME, default=user_input.get(CONF_NAME)): str,
+            vol.Required(
+                CONF_STILL_IMAGE_URL,
+                default=user_input.get(CONF_STILL_IMAGE_URL),
+            ): str,
+            vol.Optional(
+                CONF_STREAM_SOURCE,
+                default=user_input.get(CONF_STREAM_SOURCE, ""),
+            ): str,
+            vol.Optional(
+                CONF_RTSP_TRANSPORT, default=user_input.get(CONF_RTSP_TRANSPORT, None)
+            ): vol.In([None, "tcp", "udp", "udp_multicast", "http"]),
+            vol.Optional(
+                CONF_AUTHENTICATION, default=user_input.get(CONF_AUTHENTICATION)
+            ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+            vol.Optional(
+                CONF_USERNAME,
+                default=user_input.get(CONF_USERNAME, ""),
```

Comment:
Why do some of these default to an empty string versus the others which will default to `None`?

Reply:
I know that sometimes for things like username/password you have to default to `""` otherwise you get the `User input malformed` error.

Reply:
Pretty sure any optional string needs to default to `""`.

Reply:
@firstof9 Correct.  If allow username to default to None, there is no error initially, but the GUI displays 'user input malformed' when the user tries to proceed without entering a username.

Reply:
Ok so None works for everything else?

Reply:
Anything not a `str` yes.